oem/ibm/libpldmresponder/oem_ibm_handler.cpp

```diff
@@ -1019,20 +1140,32 @@ int pldm::responder::oem_ibm_platform::Handler::setNumericEffecter(
                                           interface, "CreateDump");
 
         std::map<std::string, std::variant<std::string, uint64_t>> createParams;
-        createParams["com.ibm.Dump.Create.CreateParameters.DumpType"] =
-            "com.ibm.Dump.Create.DumpType.SBE";
-        createParams["com.ibm.Dump.Create.CreateParameters.ErrorLogId"] =
-            (uint64_t)value;
-        createParams["com.ibm.Dump.Create.CreateParameters.FailingUnitId"] =
-            (uint64_t)entityInstance;
+        if (entityType == PLDM_ENTITY_MEMORY_MODULE)
+        {
+            createParams["com.ibm.Dump.Create.CreateParameters.DumpType"] =
+                "com.ibm.Dump.Create.DumpType.MemoryBufferSBE";
+            createParams["com.ibm.Dump.Create.CreateParameters.ErrorLogId"] =
+                (uint64_t)value;
+            createParams["com.ibm.Dump.Create.CreateParameters.FailingUnitId"] =
+                (uint64_t)entityInstance;
+        }
+        else
```

Comment:
Do an explicit PROC entity check here. Return error if entity is proc or dimm.

Also you could avoid code duplication:

if (dimm) {
    set dump type as MemoryBufferSBE
}
else if (proc) {
    set dump type as MemoryBufferSBE
}
else {
    retutn error
}

set ErrorLogId
set FailingUnitId

Reply:
Done